File: frontend/src/api/ocr.ts
import apiClient from "./client";

export type OcrSourceType = "HANDWRITING" | "INSURANCE_CARD" | "FORMULARY_PDF";

export interface OcrRequest {
  base64_data: string;
  mime_type: string;
  source_type: OcrSourceType;
}

export interface OcrResponse {
  sourceType: string;
  result: Record<string, unknown>;
}

export async function processOcr(payload: OcrRequest): Promise<OcrResponse> {
  return await apiClient.post("/ocr", payload) as unknown as OcrResponse;
}

File: backend/pharmasense/schemas/ocr.py
"""OCR request / response DTOs (Part 2A §5.4, §6.1)."""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, Field


class OcrRequest(BaseModel):
    base64_data: str = Field(min_length=1, description="Base64-encoded image or PDF data")
    mime_type: str = Field(description="e.g. image/png, image/jpeg, application/pdf")
    source_type: Literal["HANDWRITING", "INSURANCE_CARD", "FORMULARY_PDF"]


class OcrResponse(BaseModel):
    source_type: str
    result: dict[str, Any] = Field(description="Parsed output from the appropriate Gemini extraction")
